uproot/write/objects/util.py

```diff
@@ -15,7 +15,7 @@ def __init__(self):
 
     def _putclass(self, cursor, obj, keycursor, beg):
         start = cursor.index - keycursor.index
-        beg = beg - keycursor.index - 8
+        beg = beg - keycursor.index
```

Comment:
This is the key thing. I'm just glad this makes so much sense! (I could imagine a situation in which we twiddle something, it all works again, and we just hold our breaths. Really understanding where this came from is a huge benefit.)